### ISSUE TITLE
Change output layer activation function

### ISSUE THREAD

**Issue by TheFish18** [ultralytics/ultralytics#4494]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, does anyone know how to change the activation function on the output layer? I've looked everywhere online and in the code and can't seem to find it.

**Comment by shashichilappagari** [comment]
@TheFish18 can you please explain what you mean by activation function on output layer? Do you want to change the activation of convolution from SiLU to some other function?

**Comment by TheFish18** [comment]
![image](https://github.com/ultralytics/ultralytics/assets/142535992/b1db033b-64bc-4009-8e89-106d63a3fc3a)
(A Comprehensive Review of YOLO: From YOLOv1 and Beyond, Terven et al.)

I would like to change/customize the sigmoid and softmax functions that are used as the activation function for the output layer.

I do not want to change the activation of convolution from SiLU.

Thanks.

**Comment by glenn-jocher** [comment]
Hi @TheFish18, 

The activation functions in the output layer of the YOLOv8 model are fundamental to how the model operates and interprets its outputs. They are integrated at a low level within the architecture and cannot be easily modified without potentially significant changes to the model's prediction characteristics and effectiveness.

Sigmoid function is used for objectness score as it restricts the output to the range (0,1) providing a probability for the presence of an object in the bounding box. Similarly, softmax is used for the class prediction ensuring a probabilistic distribution over the class labels.

While it's technically possible to change these functions, doing so would require a deep understanding of their role within the YOLOv8 algorithm and a solid reasoning for why an alternative activation function might yield better results. In addition to changing the model's code, this would also likely require changing the corresponding loss functions used during training to ensure that the model's learning corresponds to the new output activation.

It would be interesting to hear more about your approach and what you hope to achieve by changing these functions! Perhaps further discussion could help elaborate on your specific needs and goals.
